go-controller/pkg/ovn/kubevirt.go

```diff
@@ -56,18 +64,300 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+		}); err != nil {
+			return fmt.Errorf("failed deleting stale policies: %v", err)
+		}
+		if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(route *nbdb.LogicalRouterStaticRoute) bool {
+			routeVMLabel, ok := route.ExternalIDs[kubevirt.VMLabel]
+			return ok && routeVMLabel == pod.Labels[kubevirt.VMLabel]
+		}); err != nil {
+			return fmt.Errorf("failed deleting stale static routes: %v", err)
+		}
+		return nil
+	}
+
+	nodeGwAddressIPv4, nodeGwAddressIPv6, err := oc.lrpAddresses(types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + pod.Spec.NodeName)
+	if err != nil {
+		return fmt.Errorf("failed reading LRP addresses: %v", err)
+	}
+
+	for _, podIP := range podAnnotation.IPs {
+		// Add a reroute policy to route VM n/s traffic to the node where the VM
+		// is running
+		ipVersion := "4"
+		nexthop := nodeGwAddressIPv4
+		if utilnet.IsIPv6CIDR(podIP) {
+			ipVersion = "6"
+			nexthop = nodeGwAddressIPv6
+		}
+		podAddress := podIP.IP.String()
+		match := fmt.Sprintf("ip%s.src == %s", ipVersion, podAddress)
+		egressPolicy := nbdb.LogicalRouterPolicy{
+			Match:    match,
+			Action:   nbdb.LogicalRouterPolicyActionReroute,
+			Nexthops: []string{nexthop},
+			Priority: 1,
+			ExternalIDs: map[string]string{
+				"namespace":      pod.Namespace,
+				kubevirt.VMLabel: pod.Labels[kubevirt.VMLabel],
+			},
+		}
+		if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(oc.nbClient, types.OVNClusterRouter, &egressPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
+			return item.Priority == egressPolicy.Priority && item.Match == egressPolicy.Match && item.Action == egressPolicy.Action
+		}); err != nil {
+			return fmt.Errorf("failed adding point to point policy: %v", err)
+		}
+
+		// Add a policy to force send an ARP to discover VMs MAC and send
+		// directly to it since there is no more routers in the middle
+		outputPort := types.RouterToSwitchPrefix + pod.Spec.NodeName
+		ingressRoute := nbdb.LogicalRouterStaticRoute{
+			IPPrefix:   podAddress,
+			Nexthop:    podAddress,
+			Policy:     &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			OutputPort: &outputPort,
+			ExternalIDs: map[string]string{
+				"namespace":      pod.Namespace,
+				kubevirt.VMLabel: pod.Labels[kubevirt.VMLabel],
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &ingressRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == ingressRoute.IPPrefix && item.Nexthop == ingressRoute.Nexthop && item.Policy != nil && *item.Policy == *ingressRoute.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route: %v", err)
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureARPProxyContainsSubnet(switchName string, subnet *net.IPNet) error {
+	lspName := ovntypes.SwitchToRouterPrefix + switchName
```

Comment:
maybe rename `lspName` -> `routerLSPName` ?

Reply:
This logic has being removed since we don't need it anymore